website/docs/BestPractices.md

```diff
@@ -90,6 +90,31 @@ It achieves this by automatically retrying the assertion until it passes or time
 await expect(button).toBeDisplayed()
 ```
 
+## Lazy loading and promise chaining
+
+WebdriverIO has some tricks up it's sleeve when it comes to writing clean code as it can lazy load the element which allows you to chain your promises and reduces the amount of `await`. This also allows you to pass the element as a ChainablePromiseElement instead of an Element and for easier use with page objects.
+
+So when do you have to use `await`?
+- whenever you perform an action using the `browser` global object such as `execute()`, `getUrl()`, etc
+- whenever you perform an action using the `element` global object, such as `click()`, `waitForDisplayed()`, etc
```

Comment:
I would phrase this the other way around: you only don't have to use `await` if fetching elements via `$`. All other cases require it.

Reply:
You're right, I'll update it. Thanks!

My team also had some nice feedback and asked if we can update our types for the `$` and `$$` command because of this as currently it requires us to `await` them according to the typees. I have a feeling we can but am curious about your view on this.

Reply:
Actually this is a fantastic idea I didn't think about before. This would help us to resolve this nasty DX bug where the IDE would autocomplete to something like this:

```ts
await (await $('foo')).click()
```

So I am 💯  for this!